File: ex-04/ex-04_c++/C3.cpp
#include "C3.hpp"

namespace p2
{

  // Constructor
  C3::C3() : a3(0)
  {
  }

  // Destructors
  C3::~C3()
  {
  }

  void C3::executar()
  {
    p1::C1 c1;
    p1::C2 c2;
    C4 c4;

    c1.setA1(1);
    c2.setA2(1);
    c4.setA4(1);

    c1.getA1();
    c2.getA2();
    c4.getA4();
  }

  // Getters & Setters
  void C3::setA3(const int &a3) { this->a3 = a3; }
  const int &C3::getA3() const { return this->a3; }
}

File: ex-04/ex-04_java/p2/C3.java
package p2;


import p1.C1;
import p1.C2;

public class C3
{

    public C3() {
        
    }
    
    public void executar()
    {
        C1 objc1 = new C1(30);
        C2 objc2 = new C2(40);
        C4 objc4 = new C4(50);
        System.out.println("Valor de obj c1 criado: " + objc1.geta1());
        objc1.seta1(20);
        System.out.println("Valor de obj c1 depois de um set: " + objc1.geta1());
        System.out.println("Valor de obj c2 criado: " + objc2.geta2());
        objc2.seta2(20);
        System.out.println("Valor de obj c2 depois de um set: " + objc2.geta2());
        System.out.println("Valor de obj c4 criado: " + objc4.geta4());
        objc4.seta4(20);
        System.out.println("Valor de obj c4 depois de um set: " + objc4.geta4());


    }

}
